sf.py

```diff
@@ -332,10 +331,20 @@ def handle_abort(signal, frame):
         if args.x and args.t:
             cfg['__outputfilter'] = args.t.split(",")
 
-        t = SpiderFootScanner(target, target, targetType, scanId,
-            modlist, cfg, dict())
-        t.daemon = True
-        t.run()
+        try:
+            s = SpiderFootScanner(target, target, targetType, scanId, modlist, cfg)
+            scanId = s.getId()
```

Comment:
Possible race: at this stage the ID might not be available.

Reply:
Disagree. The ID is set during `SpiderFootScanner` initialisation. This line is called after the `SpiderFootScanner` object is created.